pkg/agent/controller/networkpolicy/fqdn.go

```diff
@@ -822,7 +833,7 @@ func (f *fqdnController) sendDNSPacketout(pktIn *ofctrl.PacketIn) error {
 			pktIn.Data.HWDst.String(),
 			srcIP,
 			dstIP,
-			uint32(config.HostGatewayOFPort),
+			inPort,
```

Comment:
I tried it, but it may introduce flaky failure on FQDN e2e tests. I doubt there might be some other conflicts in the existing OpenFlow entries. I want to use another change to investigate it.

Reply:
Did the failure happened when it tried to access "maps.google.com"? I met it a few times and still investigating the reason. It should be not related to inport.

Reply:
Not only "maps.google.com", I hit failures with case `ACNPFQDNPolicy` and `FQDNPolicyInCluster` if I use the original packetIn message's in_port in the packet-out message. Following is the logs for failure:
```
antreapolicy_test.go:2509: failure -- wrong results for probe: Source x-ih4zhfhq/a --> Dest drive.google.com:80 connectivity: Err, expected: Rej
antreapolicy_test.go:2509: failure -- wrong results for probe: Source x-ih4zhfhq/b --> Dest maps.google.com:80 connectivity: Drp, expected: Rej
antreapolicy_test.go:2509: failure -- wrong results for probe: Source y-ih4zhfhq/b --> Dest facebook.com:80 connectivity: Drp, expected: Con
antreapolicy_test.go:2600: failure -- wrong results for probe: Source y-ih4zhfhq/b --> Dest ipv4-svc.x-ih4zhfhq.svc.cluster.local.:80 connectivity: Err, expected: Rej
antreapolicy_test.go:2600: failure -- wrong results for probe: Source z-ih4zhfhq/c --> Dest ipv4-svc.x-ih4zhfhq.svc.cluster.local.:80 connectivity: Err, expected: Drp
```

I agree the failure should be not about in_port, but I have to say using gw port as in_port, the failure is almost gone in my latest tests. So I think it is better to use another patch to resolve this failure, and I don't want to block e2e tests after this one is merged.